cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_repositories.go

```diff
@@ -33,20 +34,42 @@ func (s *Server) AddPackageRepository(ctx context.Context, request *corev1.AddPa
 	if cluster != s.globalPackagingCluster {
 		return nil, status.Errorf(codes.InvalidArgument, "installing package repositories in other clusters in not supported yet")
 	}
-	if err := s.validatePackageRepositoryCreate(request); err != nil {
+	if err := s.validatePackageRepositoryCreate(ctx, cluster, request); err != nil {
 		return nil, err
 	}
 
-	// build repository
-	repository, err := s.buildPkgRepositoryCreate(request)
+	// create secret (must be done first, to get the name)
+	var err error
+	var pkgSecret *k8scorev1.Secret
+	if request.Auth != nil && request.Auth.GetSecretRef() == nil {
+		pkgSecret, err = s.buildPkgRepositorySecretCreate(namespace, request.Name, request.Auth)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "unable to build the associated secret: %v", err)
+		}
+
+		pkgSecret, err = s.createSecret(ctx, cluster, pkgSecret)
+		if err != nil {
+			return nil, statuserror.FromK8sError("create", "Secret", request.Name, err)
+		}
+	}
+
+	// create repository
+	pkgRepository, err := s.buildPkgRepositoryCreate(request, pkgSecret)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "unable to build the PackageRepository: %v", err)
 	}
-
-	// persist repository
-	_, err = s.createPkgRepository(ctx, cluster, namespace, repository)
+	pkgRepository, err = s.createPkgRepository(ctx, cluster, namespace, pkgRepository)
 	if err != nil {
-		return nil, statuserror.FromK8sError("create", "PackageRepository", request.GetName(), err)
+		return nil, statuserror.FromK8sError("create", "PackageRepository", request.Name, err)
+	}
+
+	// update secret with owner reference if needed
+	if pkgSecret != nil {
+		addOwnerReference(pkgSecret, pkgRepository)
+		pkgSecret, err = s.updateSecret(ctx, cluster, pkgSecret)
```

Comment:
yes, i tested as well, and this is failing.
whether we allow to change the type will be part of the discussion.
another option to delete+create is to use the Opaque type instead of a defined type.

Reply:
in the discussion, we agreed we will disallow changing the type